core/consensus/authority/impl/authority_manager_impl.cpp

```diff
@@ -260,75 +324,96 @@ namespace kagome::authority {
     }
     node->descendants.emplace_back(std::move(new_node));
 
+    std::ignore = save();
+
     return outcome::success();
   }
 
   outcome::result<void> AuthorityManagerImpl::onConsensus(
       const primitives::ConsensusEngineId &engine_id,
       const primitives::BlockInfo &block,
       const primitives::Consensus &message) {
-    if (std::find(known_engines.begin(), known_engines.end(), engine_id)
-        == known_engines.end()) {
+    OUTCOME_TRY(message.decode());
+    if (engine_id == primitives::kBabeEngineId) {
+      // TODO(xDimon): Perhaps it needs to be refactored.
```

Comment:
Are you going to refactor it in this PR? If no, please add issue number.

Reply:
Issue link added